components/persistor/lib/knex/update.ts

```diff
@@ -158,7 +157,7 @@ module.exports = function (PersistObjectTemplate) {
                     const documentBody = remoteObject;
 
                     // unique identifier to find the object we're saving in the remote store
-                    const objectKey = defineProperty.__remoteObjectKey__ || `${defineProperty.remoteKeyBase}-${uniqueIdentifier}`;
+                    const objectKey = `${defineProperty.remoteKeyBase}-${uniqueIdentifier}`;
```

Comment:
nit: consider using a `/` to create folders for each property.

```suggestion
                    const objectKey = `${defineProperty.remoteKeyBase}/${uniqueIdentifier}`;
```

Reply:
I agree with this. posting what we talked about in slack for visibility - right now the logic is applied to both local and S3 key generation since this currently lives in a common piece of code, which is causing issues. follow up TODO for this would be to move this key creation logic into the remote client classes and have them be responsible for their own naming conventions given the remoteKeyBase and a unique Identifier.